### ISSUE TITLE
Can single user be a member of multiple groups?

### ISSUE THREAD

**Issue by visortelle** [keygen-sh/keygen-api#748]

**Comment by ezekg** [comment]
No. Currently, resources can belong to a single group. What's your use case here?

**Comment by visortelle** [comment]
In my case, one user account may belong to several different organizations.

It's similar to how organizations work on GitHub. You can be a member of the `keygen-sh` organization and at the same time be a member of the `microsoft` organization.

Each organization may buy products on a per-seat or metered model, e.g. increase organization members count limit or buy additional GitHub Actions minutes.

I supposed that I can implement this way:
- For each organization, I create a corresponding Keygen Group.
- For each user, I create a corresponding Keygen User and make him a member of corresponding Keygen Groups. As we figured out, currently I can't do it this way.
- When the organization admin buys a license for 10 users, I create a Keygen License with `maxMachines` set to `10`, and associate it with the corresponding Keygen Group.

@ezekg I'm new to Keygen and maybe I'm missing something.

**Comment by ezekg** [comment]
Currently, this type of grouping is not possible. You'll probably need to accomplish this grouping outside of Keygen, using user metadata as a lookup table instead of using the group relationship. Updating the issue to reflect the feature request.
